### PR TITLE
neovim: update to 0.5.0

### PR DESCRIPTION
**Pull request by queensferryme**
#### Description

* refer to <https://github.com/Homebrew/homebrew-core/commit/a508a5152f7c35284cd7276f5efbb991a054e78f>

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 11.4 20F71 x86_64
Xcode 12.5.1 12E507

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

图片

Some plugins ([nvim-tree.lua](https://github.com/kyazdani42/nvim-tree.lua), [galaxyline](https://github.com/glepnir/galaxyline.nvim), [barbar.nvim](https://github.com/romgrk/barbar.nvim), [nvim-treesitter](https://github.com/nvim-treesitter/nvim-treesitter)) for neovim 0.5.0+ only works great on my machine.

图片

### CODE REVIEW

File: editors/neovim/Portfile [hunk 1]
```diff
@@ -37,7 +35,8 @@ depends_lib             port:gettext \
                         path:lib/libluajit-5.1.2.dylib:luajit \
                         port:lua51-lpeg \
                         port:lua51-mpack \
-                        port:luv-luajit
+                        port:luv-luajit \
+                        port:tree-sitter
```
Comment (l2dy) [inline editors/neovim/Portfile:39]: Can you post the output of `otool -L /opt/local/bin/nvim`? Last time I checked, nvim is linked to libiconv.2.dylib from `port:libiconv`.

Comment (queensferryme) [reply editors/neovim/Portfile:39]:
```
/opt/local/bin/nvim:
	/System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1775.118.101)
	/opt/local/lib/libintl.8.dylib (compatibility version 10.0.0, current version 10.5.0)
	/opt/local/lib/libiconv.2.dylib (compatibility version 9.0.0, current version 9.1.0)
	/opt/local/lib/libluv.1.dylib (compatibility version 1.0.0, current version 1.41.0)
	/opt/local/lib/libuv.1.dylib (compatibility version 2.0.0, current version 2.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1292.100.5)
	/opt/local/lib/libmsgpackc.2.dylib (compatibility version 2.0.0, current version 2.0.0)
	/opt/local/lib/libvterm.0.dylib (compatibility version 1.0.0, current version 1.4.0)
	/opt/local/lib/libtermkey.1.dylib (compatibility version 16.0.0, current version 16.2.0)
	/opt/local/lib/libunibilium.4.dylib (compatibility version 5.0.0, current version 5.0.0)
	/opt/local/lib/libtree-sitter.0.dylib (compatibility version 0.0.0, current version 0.0.0)
	/usr/lib/libutil.dylib (compatibility version 1.0.0, current version 1.0.0)
	/opt/local/lib/libluajit-5.1.2.dylib (compatibility version 2.1.0, current version 2.1.0)
```

Comment (queensferryme) [reply editors/neovim/Portfile:39]: It seems `port:libiconv` is depended by `port:gettext`, which is then depended by `port:neovim`.

https://github.com/macports/macports-ports/blob/689c36329bc442db0a62516aa645297a9e92019b/devel/gettext/Portfile#L50

Comment (l2dy) [reply editors/neovim/Portfile:39]: > It seems `port:libiconv` is depended by `port:gettext`, which is then depended by `port:neovim`.

Yes, but for the sake of revision bump convenience, please add all linked libraries to depends_lib.

**Review by l2dy — APPROVED** [review]

**Comment by l2dy** [comment]
Merged. Thanks!

**Comment by raimue** [comment]
Thank you for the submission @queensferryme and thank you for handling the merge @l2dy!

**Pull request merged**